src/core/segment_buffers/implementations/text/html/html_text_segment_buffer.ts

```diff
@@ -180,11 +180,17 @@ export default class HTMLTextSegmentBuffer extends SegmentBuffer {
           this._disableCurrentCues();
           return;
         }
+        const videoElt = this._videoElement;
         const { MAXIMUM_HTML_TEXT_TRACK_UPDATE_INTERVAL } = config.getCurrent();
-        // to spread the time error, we divide the regular chosen interval.
-        const time = Math.max(this._videoElement.currentTime +
-                              (MAXIMUM_HTML_TEXT_TRACK_UPDATE_INTERVAL / 1000) / 2,
-                              0);
+        let time;
+        if (videoElt.paused || videoElt.playbackRate <= 0) {
+          time = videoElt.currentTime;
```

Comment:
I noticed that it was better to consider the exact time when paused, wheras in other cases, keeping to spread the potential error is a good idea in the vast majority of cases.